rest-api-spec/src/main/resources/rest-api-spec/test/cluster.health/10_basic.yml

```diff
@@ -92,6 +92,37 @@
   - match:     { unassigned_shards:       0 }
   - gte:       { number_of_pending_tasks: 0 }
 
+---
+"cluster health basic test, one index with wait for no initializing shards":
+  - skip:
+      version: " - 6.99.99"
+      reason: "wait_for_no_initializing_shard is introduced in 7.0.0"
+
+  - do:
+      indices.create:
+        index: test_index
+        wait_for_active_shards: 0
+        body:
+          settings:
+            index:
+              number_of_shards: 50
+              number_of_replicas: 0
+
+  - do:
+      cluster.health:
+        wait_for_no_initializing_shards: true
+
+  - is_true:   cluster_name
+  - is_false:  timed_out
+  - gte:       { number_of_nodes:         1 }
+  - gte:       { number_of_data_nodes:    1 }
+  - match:     { active_primary_shards:   50}
+  - gt:        { active_shards:           0 }
+  - gte:       { relocating_shards:       0 }
+  - match:     { initializing_shards:     0 }
```

Comment:
I think it's good enough to just test this match. The other ones are covered by other tests in this same file.